recipes/boost/all/conanfile.py

```diff
@@ -852,102 +914,126 @@ def _is_versioned_layout(self):
-                    missing_order_info.append(real_lib_name)  # Assume they do not depend on other
+                self.cpp_info.components["headers"].includedirs = [os.path.join(self.package_folder, "include", boost_version_tag)]
 
-        # Flat the list and append the missing order
-        self.cpp_info.libs = [item for sublist in ordered_libs
-                                      for item in sublist if sublist] + missing_order_info
+        self.cpp_info.components["headers"].libs = []
+        self.cpp_info.components["headers"].names["cmake_find_package"] = "boost"
+        self.cpp_info.components["headers"].names["cmake_find_package_multi"] = "boost"
 
-        if self.options.without_test:  # remove boost_unit_test_framework
-            self.cpp_info.libs = [lib for lib in self.cpp_info.libs if "unit_test" not in lib]
+        libformatdata = {}
+        if not self.options.without_python:
+            pyversion = tools.Version(self._python_version)
+            libformatdata["py_major"] = pyversion.major
+            libformatdata["py_minor"] = pyversion.minor
+
+        modules_seen = set()
+        detected_libraries = set(tools.collect_libs(self))
+        used_libraries = set()
+        for module in self._iter_modules():
+            if self.options.get_safe("without_{}".format(module), False) or not all(d in modules_seen for d in self._dependencies["dependencies"][module]):
+                continue
+            module_libraries = [lib.format(**libformatdata) for lib in self._dependencies["libs"][module]]
+            if all(l in detected_libraries for l in module_libraries):
+                modules_seen.add(module)
+
+                used_libraries = used_libraries.union(module_libraries)
+                self.cpp_info.components[module].libs = module_libraries
+
+                self.cpp_info.components[module].requires = self._dependencies["dependencies"][module] + ["headers"]
+                self.cpp_info.components[module].names["cmake_find_package"] = module
+                self.cpp_info.components[module].names["cmake_find_package_multi"] = module
+
+        if used_libraries != detected_libraries:
+            non_used = detected_libraries.difference(used_libraries)
+            if non_used:
+                raise ConanInvalidConfiguration("These libraries were not used in conan components: {}".format(non_used))
+
+            non_existing = used_libraries.difference(detected_libraries)
+            if non_existing:
+                raise ConanInvalidConfiguration("These libraries were used, but not built: {}".format(non_existing))
 
-        self.output.info("LIBRARIES: %s" % self.cpp_info.libs)
-        self.output.info("Package folder: %s" % self.package_folder)
+        if self._zip_bzip2_requires_needed:
+            if self.options.bzip2:
+                self.cpp_info.components["headers"].requires.append("bzip2::bzip2")
```

Comment:
I think we need to add these dependencies to the actual components.
if we can encode them into the yaml, would be nice. otherwise (if it's impossible/hard/time-consuming), just hard-code for now like:
```
self.cpp_info.components["iostream"].requires.append("zlib::zlib")
```
(ofcourse, it should check both `zlib` option and `without_iostream` option)

Reply:
That was one of my questions.
Do you know what boost libraries use what dependencies?

Reply:
I do know, yes. it's not something boostdep can report, just in case, right?
1. iostream: zlib, bzip2, lzma (xz utils), zstd
2. locale: icu, iconv
3. regex: icu

Reply:
Regexes can solve anything, and it came up with the following:
```
requirements:
  iostreams:
  - zlib
  - lzma
  - zstd
  - bzip2
  locale:
  - icu
  - iconv
  python:
  - python
  regex:
  - icu
  stacktrace:
  - backtrace
```

So your list was completely correct.